### PR TITLE
gsdx: value was wrongly overwritten

### PR DESCRIPTION
**Pull request by gregory38**
Need a lots of tests to ensure at least no regression on cache behavior

Issue #332

### CODE REVIEW

**Comment by ramapcsx2** [comment]
Have you run it with a bit of logging enabled (to see if games get into the conditional)?
It looks like it should make a pretty big difference if it's really bad.

**Comment by gregory38** [comment]
I don't remember, I don't think so. I think this conditional is barely taken. I need to decipher in plain english what the code exactly does. Maybe it has only an impact on memory consumption of the cache.

**Comment by gregory38** [comment]
Hum, I don't think it would make any difference.

The idea of the code is to dirty a render target when a texture upload into GS
1/ will overlap the old render target (this patch impacts this condition)
2/ have a compatible format (note this condition is unlikely)

Otherwise you can delete the cached target (less RAM, small hash table)

If the RT isn't deleted now, it will still be deleted later (based on cache eviction policy)

Note: dirty RT means than you need to transfer GS mem to GPU texture if you want to use it.

**Comment by jobermayr** [comment]
Seems to has been successful on openSUSE/Packman Linux for ~ 2 weeks (no bug reports).

**Comment by ramapcsx2** [comment]
Considering this is the dreaded caching algorithm in GSdx, deciphering and reversing this would be top notch! Until then, you might just want to put a comment on this line for now..

**Comment by karasuhebi** [comment]
So what exactly are we needing to test here? I have some free time today to compile and test.

**Comment by gregory38** [comment]
Check no major regression. Check memory consumption on game that requires lots of texture.

**Comment by karasuhebi** [comment]
Any specific titles you can think of?

**Comment by gregory38** [comment]
No. Just test any title. I think the change is safe anyway

**Comment by desertofunknown** [comment]
Looking into it...

**Comment by gregory38** [comment]
Let's merge it.
